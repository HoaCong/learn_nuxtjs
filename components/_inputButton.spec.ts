import { shallowMount } from "@vue/test-utils";
import InputText from "primevue/inputtext";
import { describe, expect, it, vi } from "vitest";
import InputButton from "./inputButton.vue";

describe("InputButton", () => {
  let wrapper: any;

  wrapper = shallowMount(InputButton, {
    propsData: {
      modelValue: "",
      placeholder: "Enter username",
      label: "Submit",
      class: "custom-class",
      handleActions: vi.fn(),
      ariaDescribedby: "description",
    },
  });

  it("renders the component correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders props correctly", () => {
    expect(wrapper.props().modelValue).toBe("");
    expect(wrapper.props().placeholder).toBe("Enter username");
    expect(wrapper.props().label).toBe("Submit");
    expect(wrapper.props().class).toBe("custom-class");
    expect(wrapper.props().ariaDescribedby).toBe("description");
  });

  it("emits an update event when input value changes", async () => {
    const input = wrapper.findComponent(InputText);
    const newValue = "Todo List";
    await input.setValue(newValue);
    expect(wrapper.emitted("update:modelValue")).toBeTruthy();
    expect(wrapper.emitted("update:modelValue")[0][0]).toBe(newValue);
  });

  it("calls handleActions method when the button is clicked", () => {
    const button = wrapper.find("button");
    button.trigger("click");
    expect(wrapper.vm.handleActions).toHaveBeenCalled();
  });
});
