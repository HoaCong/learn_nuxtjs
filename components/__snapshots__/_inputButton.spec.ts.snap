// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputButton > renders the component correctly 1`] = `"<div class=\\"flex justify-content-between\\"><span class=\\"p-float-label flex-grow-1\\"><input-text-stub modelvalue=\\"\\" class=\\"custom-class\\" aria-describedby=\\"description\\"></input-text-stub><label>Enter username</label></span><button label=\\"Submit\\"></button></div>"`;
